src/v/redpanda/admin_server.cc

```diff
@@ -602,22 +660,24 @@ void admin_server::register_kafka_routes() {
 
           auto shard = _shard_table.local().shard_for(ntp);
           if (!shard) {
-              throw ss::httpd::not_found_exception(fmt::format(
-                "Topic partition {}:{} not found", topic, partition));
+              // This node is not a member of the raft group, redirect.
+              throw redirect_to_leader(*req, ntp);
```

Comment:
repeating the same comment from the original review that might have gotten lost in the noise:

this signature is `throw ss::future<T>`. did you mean to unwrap the future and throw T?

Reply:
Thanks for bumping, this was indeed missing a co_await, fixed.